source/NationalInstruments/jquery.flot.highlights.js

```diff
@@ -0,0 +1,210 @@
+
+                        if (this._options.highlightBars) {
+                            this._drawBarHighlight(series[j], [series[j].datapoints.points[i * ps], series[j].datapoints.points[i * ps + 1]], ctx);
+                        }
+                    }
+                }
+
+                if (this._options.highlightLines && highlightIndexes.length > 0) {
+                    this._drawLineHighlight(series[j], plot, ctx);
+                }
+            }
+
+            ctx.restore();
+        }
+
+        _drawPointHighlight(series, plot, ctx, pointIndex) {
+            const ps = series.datapoints.pointsize;
+            let x = series.datapoints.points[pointIndex * ps];
+            let y = series.datapoints.points[pointIndex * ps + 1];
+            const axisx = series.xaxis;
+            const axisy = series.yaxis;
+            const highlightColor = (typeof this._options.highlightColor === "string") ? this._options.highlightColor : $.color.parse(series.color).scale('a', 0.5).toString();
+
+            if (x < axisx.min || x > axisx.max || y < axisy.min || y > axisy.max) {
+                return;
+            }
+
+            const pointRadius = series.points.radius + series.points.lineWidth / 2;
+            ctx.lineWidth = pointRadius;
+            ctx.strokeStyle = highlightColor;
+            ctx.fillStyle = highlightColor;
+            ctx.shadowColor = 'black';
+            ctx.shadowBlur = 3;
+            ctx.shadowOffsetY = 1;
+            let radius = 1.5 * pointRadius;
+            x = axisx.p2c(x);
+            y = axisy.p2c(y);
+
+            ctx.beginPath();
+            const symbol = series.points.symbol;
+            if (symbol === 'circle') {
+                ctx.arc(x, y, radius, 0, 2 * Math.PI, false);
+            } else if (typeof symbol === 'string' && plot.drawSymbol && plot.drawSymbol[symbol]) {
+                plot.drawSymbol[symbol](ctx, x, y, radius, false);
+            }
+
+            ctx.closePath();
+            ctx.fill();
+        }
+
+        _drawLineHighlight(series, plot, ctx) {
+            const plotOffset = plot.getPlotOffset();
+            ctx.translate(-plotOffset.left, -plotOffset.top);
+            const originalWidth = series.lines.lineWidth;
+            const originalColor = series.color;
+            series.lines.lineWidth += this._options.lineWidth;
+            series.color = this._options.highlightColor;
+            $.plot.drawSeries.drawSeriesLines(series, ctx, plotOffset, plot.width(), plot.height(), plot.drawSymbol, () => this._options.highlightColor);
+            series.lines.lineWidth = originalWidth;
+            series.color = originalColor;
+            $.plot.drawSeries.drawSeriesLines(series, ctx, plotOffset, plot.width(), plot.height(), plot.drawSymbol, () => this._options.highlightColor);
+            $.plot.drawSeries.drawSeriesPoints(series, ctx, plotOffset, plot.width(), plot.height(), plot.drawSymbol, () => this._options.highlightColor);
```

Comment:
Why are we drawing points here?

Reply:
So that if the view contains lines and points they both show up above the selection

Reply:
But this is going to always draw highlighted points, regardless if no points were drawn for the regular, non-highlighted, data. I would think the _drawPointHightlight method is wholly responsible for drawing highlighted points. Perhaps this comment is in concert with our in-person discussion about _drawLineHighlight honoring the selection input.

Reply:
This call does not draw highlighted points - just regular points.

Reply:
Resolved in person.